### PR TITLE
Add thumbnail method to URLExtensions

### PR DESCRIPTION
**Pull request by BennX**
This pull request should solve #360. 
It provides a simple extension to URL to generate a thumbnail from a given URL. It returns an optional `UIImage?` of an given url. If the AVFoundation cant generate the thumbnail it'll simply return nil.

In addition the documentation informs the user that it is recommended to dispatch the function call in case it should generate a thumbnail from a remote source since it may take some time.

The test may need a different "test" url where we are sure that it wont change in the future.

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a changelog entry describing my changes.

### CODE REVIEW

File: Sources/Extensions/Foundation/URLExtensions.swift [hunk 1]
```diff
@@ -55,5 +56,34 @@ public extension URL {
...
+    ///        let image_at_0seconds = url.thumbnail()
+    ///        let image_at_5seconds = url.thumbnail(fromTime: 5)
+    ///
+    ///        DisptachQueue.main.async {
+    ///            someImageView.image = url.thumbnial()
+    ///        }
+    ///
+    /// - Parameter fromTime: Seconds into the video where the image should be generated.
+    public func thumbnail(fromTime time: Float64 = 0) -> UIImage? {
+        let asset: AVAsset = AVAsset(url: self)
```
Comment (SD10) [inline Sources/Extensions/Foundation/URLExtensions.swift]: Can we remove the explicit `AVAsset` typing and let it be inferred?

File: Sources/Extensions/Foundation/URLExtensions.swift [hunk 2]
```diff
@@ -55,5 +56,34 @@ public extension URL {
...
+        let imageGenerator = AVAssetImageGenerator(asset: asset)
+        let time = CMTimeMakeWithSeconds(time, 1)
+        
+        var actualTime: CMTime = CMTimeMake(0, 0)
```
Comment (SD10) [inline Sources/Extensions/Foundation/URLExtensions.swift]: Same with `actualTime`

File: Sources/Extensions/Foundation/URLExtensions.swift [hunk 3]
```diff
@@ -55,5 +56,34 @@ public extension URL {
...
+// MARK: - Methods
+public extension URL {
+        
+    /// SwifterSwift: Generate a thumbnail image from given url. Returns nil if no thumbnail could be created. This function may take some time to complete. It's recommended to dispatch the call if the thumbnail is not generated from a local resource.
+    ///
+    ///        var url = URL(string: "https://video.golem.de/files/1/1/20637/wrkw0718-sd.mp4")!
+    ///        let image_at_0seconds = url.thumbnail()
+    ///        let image_at_5seconds = url.thumbnail(fromTime: 5)
+    ///
+    ///        DisptachQueue.main.async {
```
Comment (SD10) [inline Sources/Extensions/Foundation/URLExtensions.swift]: Let's remove the example of assigning here. Should be self explanatory

File: Sources/Extensions/Foundation/URLExtensions.swift (hunk 2 above)
Comment (BennX) [reply Sources/Extensions/Foundation/URLExtensions.swift]: Sure 👍 However that slipped in the code.

**Comment by BennX** [comment]
I'm confused i didn't close this PR.

**Pull request closed without merging**